### PR TITLE
Voting Mechanic

### PR DESCRIPTION
**Pull request by legojoshua12**
Adds voting from dead players to give an extra action a day to a player

### CODE REVIEW

File: libraries/boardConstructor.py [hunk 1]
```diff
@@ -1,35 +1,52 @@
...
 
 
 def populateBoard(board, players):
+    """
+    Inserts each player on the board in random positions while constructing a new board
+    :param board: The partially completed board array
+    :param players: The number of players in the game
+    """
     for player in range(players):
         while True:
```
Comment (AndrewLao) [inline libraries/boardConstructor.py:27]: probably replace while True with an equivalent if statement for conciseness

File: main.py [hunk 2]
```diff
@@ -67,13 +86,81 @@ async def on_reaction_add(reaction, user):
...
+                            elif int(data[server][channel]['players'][player]['votes']) == championVotes:
+                                champions += 1
+                            else:
+                                data[server][channel]['players'][player]['votes'] = 0
+                        data[server][channel]['players'][player]['votes'] = 0
+                # TODO add messages here in the channel saying if there was a winner, loser, or tie
+                if champions == 1:
+                    data[server][channel]['players'][championPlayer]['actions'] = int(data[server][channel]['players']
+                                                                                      [championPlayer]['actions']) + 1
+                elif champions >= 2:
```
Comment (AndrewLao) [inline main.py:146]: isn't the elif and else pass redundant

File: libraries/messageHandler.py [hunk 3]
```diff
@@ -99,12 +111,16 @@ async def handleMessage(message, client, commandMessageStarter):
                     await commands.displayBoard(message, renderedBoard)
                 elif action == 'players':
                     await commands.showPlayerStatistics(message, jsonManager.readJson(), client)
+                elif action == 'dm':
```
Comment (AndrewLao) [inline libraries/messageHandler.py:114]: wait is this copy-paste from line 66

**Review by AndrewLao — COMMENTED** [review]
There's some stuff you might be able to clean up in commands to make things look more concise and I put a couple of comments here and there. No major efficiency issues I could see.

File: libraries/messageHandler.py (hunk 3 above)
Comment (legojoshua12) [reply libraries/messageHandler.py:114]: Yes, it is the same as line 66 as one is a command for an already started game and one is a command from the lobby, but both do the same action as of now

File: main.py (hunk 2 above)
Comment (legojoshua12) [reply main.py]: Fixed

**Pull request merged**